### ISSUE TITLE
Cannot use bitwise and

### ISSUE THREAD

**Issue by stefandrissen** [simonowen/pyz80#17]
While I can use a bitwise OR to enable the RAM bit on my page:
````
low.memory.ram.0:                   equ %00100000
page.main:              equ 2

ld a,page.main | low.memory.ram.0

````
I cannot use a bitwise AND to mask off the page:
````
page.heap:              equ 0   ; basic
low.memory.page.mask:               equ %00011111

@section.B.heap: equ ( page.heap - 1 ) & low.memory.page.mask
````
Assembly fails as a result of Comet compatiblity - & is replaced with 0x:
````
    arg = arg.replace('&','0x') # COMET syntax for hex numbers
````
To allow a bitwise AND this chould be a bit less blunt, something like (note that this does not work):
````
arg = re.sub('&([0-9a-fA-F]+)', '0x\g<1>', arg)
````

**Comment by simonowen** [comment]
Your last example seems better than the existing `arg.replace`, though it needs a little more tweaking to not match some unwanted cases. It's difficult to tell unary prefix from binary operator without a full expression parser, but it should be possible to match well enough that most people won't notice.

I'm hoping an expanded regular expression can be made to work. Worst-case is probably a re.findall to iterate over possible matches for extra checking. I'll experiment...

**Comment by simonowen** [comment]
I've committed something pretty close to your last expression, just with an extra word-break check. It should allow bitwise and to be used in most cases, and at worst you'll need a space after the `&` if it's being mistaken for a hex value.

I think correctly identifying whether it's a unary or binary operator would require an expression parser, and that doesn't fit with how the current version uses Python for expression evaluation. It may be possible to improve the regex a little, but in my tests I was always able to find an expression that could break anything I wrote.
